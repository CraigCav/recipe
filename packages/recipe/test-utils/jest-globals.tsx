import React from 'react';
import {render, cleanup} from '@testing-library/react';
import {ThemeProvider} from 'emotion-theming';
import {standard} from '../themes';

import '@testing-library/jest-dom/extend-expect';

const renderWithTheme = renderFn => (component, ...rest) =>
  renderFn(<ThemeProvider theme={standard}>{component}</ThemeProvider>, rest);

export const renderToHtml = renderWithTheme((ui, options) => {
  const {container} = render(ui, options);
  return container.outerHTML;
});

export const fullRender = renderWithTheme(render);

afterEach(cleanup);
