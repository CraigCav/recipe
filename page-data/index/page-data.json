{"componentChunkName":"component---src-pages-index-js","path":"/","webpackCompilationHash":"356c1bd9f0f7a983bfed","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}