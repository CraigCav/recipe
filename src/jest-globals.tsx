import {shallow as enzymeShallow, render as enzymeRender, mount as enzymeMount} from 'enzyme';
import React from 'react';
import {create as reactCreate} from 'react-test-renderer';
import {render as rtlRender, cleanup} from 'react-testing-library';
import {renderToStaticMarkup} from 'react-dom/server';
import {ThemeProvider} from 'emotion-theming';
import {standard} from './themes';

import '@testing-library/jest-dom/extend-expect';

const renderWithTheme = renderFn => (component, ...rest) =>
  renderFn(<ThemeProvider theme={standard}>{component}</ThemeProvider>, rest);

export const shallow = tree => {
  const wrapper = enzymeShallow(<ThemeProvider theme={standard} />);
  const instance = wrapper.instance() as any;
  const context = instance.getChildContext();

  return enzymeShallow(tree, {context});
};

export const mount = tree => {
  const wrapper = enzymeShallow(<ThemeProvider theme={standard} />);
  const instance = wrapper.instance() as any;
  const context = instance.getChildContext();

  return enzymeMount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

export const renderToHtml = renderWithTheme(renderToStaticMarkup);

export const render = renderWithTheme(enzymeRender);

export const fullRender = renderWithTheme(rtlRender);

export const create = renderWithTheme(reactCreate);

afterEach(cleanup);
