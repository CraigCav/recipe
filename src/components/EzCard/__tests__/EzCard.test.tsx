import React from 'react';
import {EzCard, EzCardSection, EzCardExpandable} from '..';
import {axe} from 'jest-axe';
import {visualSnapshots} from 'sosia';
import {renderToStaticMarkup} from 'react-dom/server';
import {renderToHtml} from '../../../jest-globals';
import regressionTests from './EzCard.test.md';
import markdown from '../EzCard.md';
import {EzPage, EzPageSection, EzLayout, EzButton} from '../../index';

const scope = {EzButton, EzCard, EzCardSection, EzCardExpandable, EzLayout, EzPage, EzPageSection};

describe('EzCard', () => {
  visualSnapshots({markdown, scope});
  visualSnapshots({markdown: regressionTests, scope});

  it('should NOT throw if no ThemeProvider is configured', () => {
    const actual = renderToStaticMarkup(
      <EzCard>
        <p>Lorem ipsum dolor</p>
      </EzCard>
    );
    expect(actual).toContain('Lorem ipsum dolor');
  });

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(
      <EzCard title="Card Heading">
        <EzCardSection>Lorem ipsum dolor</EzCardSection>
        <EzCardSection>sit amet, consectetur adipiscing</EzCardSection>
      </EzCard>
    );
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });

  describe('data-* props', () => {
    let actual;
    beforeEach(() => {
      actual = renderToHtml(
        <EzCard data-test="my-test-selector">
          <EzCardSection>content</EzCardSection>
        </EzCard>
      );
    });
    it('renders valid props for html elements', () => {
      expect(actual).toContain('data-test="my-test-selector"');
    });
  });
});
