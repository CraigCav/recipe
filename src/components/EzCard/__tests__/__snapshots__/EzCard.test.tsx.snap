// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzCard should render card sections horizontally 1`] = `
.recipe-6 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.recipe-4 > * {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className="recipe-6 recipe-7"
>
  <div
    className="recipe-4 recipe-5"
  >
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
  </div>
</div>
`;

exports[`EzCard should render with card sections 1`] = `
.recipe-6 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
}

.recipe-4 > *:not(:first-child) {
  border-top: 1px solid #ced4d9;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

<div
  className="recipe-6 recipe-7"
>
  <div
    className="recipe-4 recipe-5"
  >
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
    <div
      className="recipe-0 recipe-1"
    >
      Lorem ipsum dolor
    </div>
  </div>
</div>
`;

exports[`EzCard should render with default styles 1`] = `
.recipe-4 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
}

.recipe-2 > *:not(:first-child) {
  border-top: 1px solid #ced4d9;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

<div
  className="recipe-4 recipe-5"
>
  <div
    className="recipe-2 recipe-3"
  >
    <div
      className="recipe-0 recipe-1"
    >
      <p>
        Lorem ipsum dolor
      </p>
    </div>
  </div>
</div>
`;

exports[`EzCard should render with header and subheader styles 1`] = `
.recipe-11 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
}

.recipe-9 > *:not(:first-child) {
  border-top: 1px solid #ced4d9;
}

.recipe-7 {
  padding: 1.4285714285714286rem;
}

.recipe-7 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-5 {
  margin: 1.4285714285714286rem 1.4285714285714286rem 0;
}

.recipe-1 {
  color: #565a5c;
  font-size: 1.2857142857142858rem;
  font-weight: 700;
  line-height: 1.35;
  margin: 0;
}

.recipe-3 {
  color: #8b99a6;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.35;
  margin-top: 0.5714285714285714rem;
}

<div
  className="recipe-11 recipe-12"
>
  <div
    className="recipe-5 recipe-6"
  >
    <header>
      <h3
        className="recipe-0 recipe-1 recipe-2"
        size="3"
      >
        Card Heading
      </h3>
      <div
        className="recipe-3 recipe-4"
      >
        Card Subheading
      </div>
    </header>
  </div>
  <div
    className="recipe-9 recipe-10"
  >
    <div
      className="recipe-7 recipe-8"
    >
      <p>
        Lorem ipsum dolor
      </p>
    </div>
  </div>
</div>
`;

exports[`EzCard should render with header styles 1`] = `
.recipe-9 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
}

.recipe-7 > *:not(:first-child) {
  border-top: 1px solid #ced4d9;
}

.recipe-5 {
  padding: 1.4285714285714286rem;
}

.recipe-5 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-3 {
  margin: 1.4285714285714286rem 1.4285714285714286rem 0;
}

.recipe-1 {
  color: #565a5c;
  font-size: 1.2857142857142858rem;
  font-weight: 700;
  line-height: 1.35;
  margin: 0;
}

<div
  className="recipe-9 recipe-10"
>
  <div
    className="recipe-3 recipe-4"
  >
    <h3
      className="recipe-0 recipe-1 recipe-2"
      size="3"
    >
      Card Heading
    </h3>
  </div>
  <div
    className="recipe-7 recipe-8"
  >
    <div
      className="recipe-5 recipe-6"
    >
      <p>
        Lorem ipsum dolor
      </p>
    </div>
  </div>
</div>
`;

exports[`EzCard should render with info accent 1`] = `
.recipe-2 > *:not(:first-child) {
  border-top: 1px solid #ced4d9;
}

.recipe-0 {
  padding: 1.4285714285714286rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

.recipe-4 {
  background: #ffffff;
  border: 1px solid #ced4d9;
  border-radius: 6px;
  border-left: 5px solid #3e90d6;
}

<div
  className="recipe-4 recipe-5"
>
  <div
    className="recipe-2 recipe-3"
  >
    <div
      className="recipe-0 recipe-1"
    >
      <div>
        Lorem ipsum
      </div>
    </div>
  </div>
</div>
`;
